scripts/events.py

```diff
@@ -817,138 +827,181 @@ def other_interactions(self, cat):
         if interactions:
             game.cur_events_list.append(choice(interactions))
 
-    def handle_deaths(self, cat):
+    def handle_injuries_or_general_death(self, cat):
+        """ 
+        This function will handle:
+            - classic mode: death events with injuries
+            - expanded mode: getting a new injuries (extra function in own class)
+        Returns: 
+            - boolean if a death event occurred or not
+        """
+        #if game.clan.game_mode == 'expanded':
+            #print("TODO")
+            #if game.settings.get('disasters'):
+            #    if not random.getrandbits(10):  # 1/1024
+            #        triggered_death = True
+            #        self.handle_disasters(cat)
+            #return
+
+        # get the general information about the cat and a random other cat
+        triggered_death = False
+        name = str(cat.name)
+        other_cat = choice(list(Cat.all_cats.values()))
+        countdown = int(len(Cat.all_cats) / 3)
+        while cat == other_cat or other_cat.dead or other_cat.status == 'leader' or other_cat.exiled:
+            other_cat = choice(list(Cat.all_cats.values()))
+            countdown -= 1
+            if countdown <= 0:
+                return
+        other_name = other_cat.name
         clan_has_kits = any(
```

Comment:
Could maybe be replaced with the kits lambda from the scars/disaster functions

```python
        alive_kits = list(filter(lambda kitty: (kitty.age == "kitten"
                                                and not kitty.dead
                                                and not kitty.exiled),
                                 Cat.all_cats.values()))
```
Not sure if that's the best way to do this either but it's ~35% faster.

Might be better either way to move this down to under the 1/400 block where it's used though, there's no need to check this for every call rn since iterating through every cat is pretty intense. It takes ~20% of the entire function's runtime.

Reply:
Moved the function to 1/400 block and replaced it